app/interface/controllers/metrics/requests.py

```
@@ -1,18 +1,26 @@
 import asyncio
+import logging
+from typing import AsyncGenerator
```

Ruff (0.14.7)

Import from `collections.abc` instead: `AsyncGenerator`

Import from `collections.abc`

(UP035)